apps/checkout-app/pages/api/payment-status/[orderId].ts

```diff
@@ -0,0 +1,97 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { Types as AdyenTypes } from "@adyen/api-library";
+import { OrderStatus as MollieOrderStatus } from "@mollie/api-client";
+
+import { allowCors } from "@/checkout-app/backend/utils";
+import { getOrderPaymentDetails } from "@/checkout-app/backend/payments/getOrderPaymentDetails";
+import { OrderPaymentMetafield } from "@/checkout-app/types";
+import { verifyAdyenSession } from "@/checkout-app/backend/payments/providers/adyen/verifySession";
+import { PaymentStatusResponse } from "@/checkout-app/types/api/payment-status";
+import { verifyMollieSession } from "@/checkout-app/backend/payments/providers/mollie/verifySession";
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "GET") {
+    res.status(405).send({ message: "Only GET requests allowed" });
+    return;
+  }
+
+  const orderId = req.query.orderId as string;
+
+  const order = await getOrderPaymentDetails(orderId);
+
+  if ("errors" in order) {
+    return res.status(400).json({
+      ok: false,
+      errors: order.errors,
+    });
+  }
+
+  let response: PaymentStatusResponse = {
+    status: "UNPAID",
+  };
+
+  // INFO: This logic needs to be revisited for multi-payment flow
+  if (
+    order.data.isPaid ||
+    order.data.authorizeStatus === "FULL" ||
+    order.data.chargeStatus === "FULL"
+  ) {
+    response.status = "PAID";
+  } else if (order.data.privateMetafield) {
+    const data: OrderPaymentMetafield = JSON.parse(order.data.privateMetafield);
+
+    if (data.provider === "adyen") {
```

Comment:
Can we extract mollie and adyen parts to dedicated functions? Would make this function easier to read